content/en/blog/_posts/2025-XX-XX-nftables-kube-proxy/index.md

```diff
@@ -0,0 +1,238 @@
+underlying kernel subsystem), it's still possible that some existing
+network plugins that work with kube-proxy in iptables mode might not
+work right with it in nftables mode; you should consult the
+documentation (or forums, bug tracker, etc.) for your network plugin to
+see if it supports nftables mode. (Note that some plugins may not need
+any changes; as long as they don't try to directly interact with or
+override kube-proxy's iptables rules, they shouldn't care whether
+kube-proxy is using iptables or nftables.)
+
+Second, the nftables mode will not work on older Linux distributions;
+currently it requires a 5.13 or newer kernel. Additionally, because of
+bugs in early versions of the `nft` command line tool, you should not
+run kube-proxy in nftables mode on nodes that have a pre-"1.0.0" version
+of `nft` in the host filesystem (or else kube-proxy's use of nftables
+may interfere with other uses of nftables on the system).
+
+Finally, kube-proxy in nftables mode is intentionally not 100%
+compatible with kube-proxy in iptables mode, because some of iptables
+mode's legacy behaviors are inherently less efficient or less secure.
+(In particular, with nftables mode, NodePort services are only reachable
+on their nodes' default IPs, as opposed to being reachable on all IPs,
+including `127.0.0.1`, with iptables mode.) The [kube-proxy
+documentation] has more information about this, including information
+about metrics you can look at to determine if you are relying on any of
+the changed functionality, and what configuration options are available
+to get more backward-compatible behavior.
+
+[kube-proxy documentation]: https://kubernetes.io/docs/reference/networking/virtual-ips/#migrating-from-iptables-mode-to-nftables
+
+## Trying out NFTables Mode
+
+Ready to try it out? In Kubernetes 1.31 and later, you just need to
+pass `--proxy-mode nftables` to kube-proxy (or set `mode: nftables` in
+your kube-proxy config file).
+
+If you are using kubeadm to set up your cluster, the kubeadm
+documentation explains [how to pass a `KubeProxyConfiguration` to
+`kubeadm init`]. You can also [deploy nftables-based clusters with
+`kind`].
+
+You can also convert existing clusters from iptables (or ipvs) mode to
+nftables by updating the kube-proxy configuration and restarting the
+kube-proxy pods. (You do not need to reboot the nodes: when restarting
+in nftables mode, kube-proxy will delete any existing iptables or ipvs
+rules, and likewise, if you later revert back to iptables or ipvs
+mode, it will delete any existing nftables rules.)
+
+[how to pass a `KubeProxyConfiguration` to `kubeadm init`]: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#customizing-kube-proxy
+[deploy nftables-based clusters with `kind`]: https://kind.sigs.k8s.io/docs/user/configuration/#kube-proxy-mode
+
+## Future Plans
+
+As mentioned above, while nftables is now the _best_ kube-proxy mode,
```

Comment:
I would love to fix or at least get some fresh data on kube-proxy restart time (which is where it may not be the best yet). I am going to do that before 1.33, so maybe we can update this post later with the fresh data, or ignore this part completely?